### PR TITLE
Circle CI Successful Build Implementation

### PR DESCRIPTION
**Pull request by danra2**
Kept the default sections and related build commands including the dependency installations. Updated Node version, and integrated build commands from localbuild.sh. Left the echo commands for more granularity in debugging. Build completed successfully.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,63 +1,165 @@
...
     steps:
       - checkout
       - restore_cache:
           keys:
-          - devcenter-deps-{{ .Branch }}-{{ checksum "Gemfile" }}
-          - devcenter-deps-preview-      
-      - run: bundle install 
+            - rubygems-v1-{{ checksum "Gemfile.lock" }}
+            - rubygems-v1-fallback
+      - run: sudo apt-get install python3
```
Comment (bjs-code) [inline .circleci/config.yml:15]: I'm not convinced we still need python.  Would be interesting to see if the build still works without installing python3, pip, pipend and Unipath

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,63 +1,165 @@
...
-      - run: ./build/write_branch.sh
-      - run: ./build/write_commit.sh
-      - run: ./build/write_forms.sh
-      - run: bundle exec jekyll build
+            - vendor/bundle
+      - run:
+          name: Widdershins Install and Execution.
+          command: |
+                sudo npm install -g widdershins
+                cd ./src/api-explorer/ && find . -name \*.json -type f -exec widdershins \{\} -o \{\}.md --omitHeader \;
```
Comment (bjs-code) [inline .circleci/config.yml:37]: We don't need this here... we're doing it at 67 below

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,63 +1,165 @@
...
+      - run:
```
Comment (bjs-code) [inline .circleci/config.yml:38]: We don't need this step for circleci

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,63 +1,165 @@
...
+                rm -rf src/Slate-API-Explorer-Reference/slate/source/images
+                rm -rf src/Slate-API-Explorer-Reference/slate/source/includes
+                rm -rf src/Slate-API-Explorer-Reference/slate/build
+                rm -rf src/staged
+                mkdir _site
+                mkdir src/Slate-API-Explorer-Reference/slate/source/images
+                mkdir src/Slate-API-Explorer-Reference/slate/source/includes
+                mkdir src/Slate-API-Explorer-Reference/slate/build
+                mkdir src/staged
+                mkdir src/staged/slate-ui
```
Comment (bjs-code) [inline .circleci/config.yml:56]: Don't need this line (should be removed in localbuild.sh too)

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,63 +1,165 @@
...
+                echo "Generating static html from markdown..."
+                pushd ./src/Slate-API-Explorer-Reference/slate && bundle install && bundle exec middleman build 2>/dev/null
+                popd
+      - run:
+          name: Modify static html files to add yaml front matter, otherwise Jekyll cannot process the file.
+          command: |
+                echo "Updating static HTML files to prepare for JEKYLL"
+                node ./build-tools/reference_fix_jekyll.js
+      - run:
+          name: Move static files to slate-ui directory.
```
Comment (bjs-code) [inline .circleci/config.yml:106]: Change name to:
Move static files to staging directory.

**Review by bjs-code — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (danra2) [reply .circleci/config.yml:15]: You are right, since we don't use the python scripts anymore. Removed.

File: .circleci/config.yml (hunk 2 above)
Comment (danra2) [reply .circleci/config.yml:37]: Resolved.

File: .circleci/config.yml (hunk 3 above)
Comment (danra2) [reply .circleci/config.yml:38]: Resolved

File: .circleci/config.yml (hunk 5 above)
Comment (danra2) [reply .circleci/config.yml:106]: Done.

File: .circleci/config.yml (hunk 4 above)
Comment (danra2) [reply .circleci/config.yml:56]: Completed.

**Review by bjs-code — APPROVED** [review]

**Pull request merged**